src/main/java/org/openstreetmap/atlas/geography/atlas/raw/creation/RawAtlasGenerator.java

```diff
@@ -446,10 +488,12 @@ private Atlas rebuildAtlas(final Atlas atlas, final Set<Long> pointsToRemove,
      * other features in the Atlas.
      * <li>2. There are multiple {@link Point}s at a {@link Location}. In this case, we sort all the
      * points, keep the one with the smallest identifier, add a {@link SyntheticDuplicateOsmNodeTag}
-     * and remove the rest of the duplicate points. Note: we can potentially be tossing out OSM
-     * Nodes with non-empty tags. However, this is the most deterministic and simple way to handle
-     * this. The presence of the synthetic tag will make it easy to write an Atlas Check to resolve
-     * the data error.
+     * and remove the rest of the duplicate points. Two notes: 1. We keep Nodes if they have
+     * different layer tagging. This way, we aren't creating a false connection between an overpass
+     * and a road beneath it, which happened to have a way node at the identical location. 2. We are
+     * potentially be tossing out OSM Nodes with non-empty tags. However, this is the most
+     * deterministic and simple way to handle this. The presence of the synthetic tag will make it
+     * easy to write an Atlas Check to resolve the data error.
```

Comment:
So when `PackedAtlasBuilder` is called and two nodes are added with the same location, the builder will connect all the edges to one of the nodes and leave the other node connected to nothing. In the end it seems it would still connect that overpass and the below road in the final atlas. Let me know if there is something I misunderstood there!

Reply:
There won't be two nodes at the same location. The logic would be something like this:

1. There would be two Atlas Points at that location in the raw Atlas.
2. There would be two Atlas Points at that location in the sliced raw Atlas.
3. During way-sectioning, when determining whether a raw Atlas Point is a shape-point, an Atlas Edge or an Atlas Point - we used to make this a Node (ignored layer values) and the PackedAtlasBuilder would connect the edges to this Node. Now, we factor in layer values, and instead of the raw Atlas Point become a Node in the final Atlas, it's simply a shape-point - since this location doesn't have unique tagging, is not a self-intersection and is not at a valid intersection with an edge of the same layer. When the PackedAtlasBuilder get() is called, there is no connection that is formed at the location. Just visualized a few example of this in JOSM as well. Let me know if I missed anything!